config/v1/types_authentication.go

```diff
@@ -771,3 +831,53 @@ type TokenRequiredClaim struct {
 	// +required
 	RequiredValue string `json:"requiredValue"`
 }
+
+type TokenExpressionRule struct {
+	// expression is a CEL expression evaluated against token claims.
+	// The expression must be a non-empty string and no longer than 4096 characters.
+	// This field is required.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=4096
+	// +required
+	// +openshift:enable:FeatureGate=ExternalOIDCWithUpstreamParity
+	Expression string `json:"expression,omitempty"`
+
+	// message allows configuring the human-readable message that is returned
+	// from the Kubernetes API server when a token fails validation based on
+	// the CEL expression defined in 'expression'. This field is optional.
```

Comment:
I don't think this is quite true. Reading https://kubernetes.io/docs/reference/access-authn-authz/authentication/#using-authentication-configuration it looks like this is the message that is _logged_ by the Kubernetes API server.

It isn't included in anything returned to the caller.